File: engine/model/src/main/scala/client/database/data/GameData.scala
package client.database.data

import client.database.GameType

class GameData(private var _type: GameType, private var _wins: Int, private var _losses: Int, private var _ties: Int) {
  def theType: GameType = _type

  def wins: Int = _wins

  def losses: Int = _losses

  def ties: Int = _ties

  def incrementWins(): Unit = {
    _wins += 1
  }

  def incrementLosses(): Unit = {
    _losses += 1
  }

  def incrementTies(): Unit = {
    _ties += 1
  }
}

File: engine/engine-model/src/main/java/client/database/data/GameData.java
package client.database.data;

import client.database.GameType;

public record GameData(GameType theType, Integer wins, Integer losses, Integer ties) {
   public GameData incrementWins() {
      return new GameData(theType, wins + 1, losses, ties);
   }

   public GameData incrementLosses() {
      return new GameData(theType, wins, losses + 1, ties);
   }

   public GameData incrementTies() {
      return new GameData(theType, wins, losses, ties + 1);
   }
}
